plugins/lookup/hashi_vault.py

```diff
@@ -631,6 +704,8 @@ def validate_auth_ldap(self, auth_method):
 
     def validate_auth_approle(self, auth_method):
         self.validate_by_required_fields(auth_method, 'role_id')
+        # https://github.com/ansible-collections/community.hashi_vault/issues/35
+        self.get_option('secret_id')
```

Comment:
Not sure what this is doing here.

Reply:
See #35 
At the moment, it doesn't actually do anything, but bcoca has been pretty insistent that `get_option()` is the only "correct" way to retrieve options, and that some hooks may be missed by relying only on `set_options()`; as it relates currently, his proposed solution for the deprecation issue also would require it. So this is a hedge (more details in the linked issue).

Reply:
That doesn't really convince me :) Deprecations must be processed in `set_options()` and not only in `get_option()`, otherwise deprecations are seriously broken IMO.

But yeah, keeping this here definitely won't hurt (except it looks superfluous ;) ).

Reply:
I tend to agree, but I am accounting for all current known possibilities as best as possible! :) 
I can improve the comment with more than a link if you think there's something succinct enough that would help; as I'm sure you've noticed terseness is not my strong suit 😅 so though it'd be better to link a longer explanation.